File: DFS&BFS/python/1967.py
from collections import deque
import sys
input = sys.stdin.readline
n = int(input())
data = [[] for _ in range(n+1)]
idx = 1
for _ in range(n-1):
    a,b,c = map(int,input().split())
    data[a].append([b,c])
    data[b].append([a,c])
    
def search(tree,visited,v):
    queue = deque()
    queue.append([v,0])
    max_cost = 0
    idx = 0
    while queue:
        node,cost = queue.popleft()
        if max_cost<cost:
            idx = node
            max_cost = cost
        visited[node] = True
        for next_ in tree[node]:
            next_node = next_[0]
            next_cost = next_[1]
            if visited[next_node]:
                continue
            queue.append([next_node,cost+next_cost])
    return idx,max_cost

for _ in range(2):
    visited = [False for _ in range(n+1)]
    idx,max_cost = search(data,visited,idx)

print(max_cost)

File: DFS&BFS/javascript/1967.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
const data = [];
let idx=1 ,cost = 0;
for (let i = 0 ; i<=n ; i++){
  data.push([]);
}
// nodeNum, nodeCost;
for (let i = 1 ; i<n ; i++){
  let [a,b,c] = input[i].split(' ').map(Number);
  data[a].push([b,c]);
  data[b].push([a,c]);
}
const dfs = (startNodeInfo) =>{
  let idx,cost;
  const visited = new Array(n+1).fill(false);
  visited[startNodeInfo[0]] = true;
  const stack = [startNodeInfo];
  const result = [0,0];
  let nextIdx,nextCost;
  while (stack.length !== 0){
    [idx,cost] = stack.pop();
    if (cost > result[1]){
      result[0] = idx;
      result[1] = cost;
    }
    for (let i = 0 ; i< data[idx].length ; i++){
      [nextIdx,nextCost] = [data[idx][i][0], data[idx][i][1]];
      if (visited[nextIdx]) continue;
      stack.push([nextIdx,nextCost+cost])
      visited[nextIdx] = true;
    }
  }
  return result;
}
for (let i = 0 ; i<2 ; i++){
  cost = 0;
  [idx,cost] = dfs([idx,cost]);
}
console.log(cost);